kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -49,15 +51,48 @@ def _hash(string: str) -> str:
     return hashlib.sha512(bytes(string, encoding="utf8")).hexdigest()
 
 
-def _get_hashed_username():
+def _get_or_create_uuid():
+    """
+    Reads a UUID from a configuration file or generates and saves a new one if not present.
+    """
+    config_path = user_config_dir("kedro")
+    full_path = os.path.join(config_path, CONFIG_FILENAME)
+
     try:
-        username = getpass.getuser()
-        return _hash(username)
-    except Exception as exc:
-        logger.warning(
-            "Something went wrong with getting the username. Exception: %s",
-            exc,
-        )
+        if os.path.exists(full_path):
+            with open(full_path) as f:
+                config = toml.load(f)
+
+                if "telemetry" in config and "uuid" in config["telemetry"]:
+                    try:
```

Comment:
looks like we don't need additional try more, because we are returning the same for any error on level upper